### PR TITLE
Upload build context to S3

### LINKED ISSUE #84: Electron uploads build context
### Description of the requested feature:

The zetaforge electron app should upload a kaniko build context to S3.

### Brief explanation on why you want/need said feature:

It is required in cloud deployment of Anvil to build the images with kaniko.

### Do you intend on coding this feature yourself?

Yes

### Which zetaforge components would be affected by this new feature:

- [X] client
- [X] anvil
- [ ] driver

### Share any code snippets related to the feature here(optional):

_No response_

### Provide an API design for the feature(optional):

_No response_

### PR DESCRIPTION
**Pull request by FGRCL**
Closes #84

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -283,6 +285,26 @@ func main() {
 			ctx.String(err.Status(), err.Error())
 		}
 	})
+	router.GET("/config", func(ctx *gin.Context) {
```
Comment (plparent) [inline main.go:288]: I don't think that the client should rely on the backend to know if it should be running locally.

File: main.go [hunk 2]
```diff
@@ -283,6 +285,26 @@ func main() {
...
+		}
+
+		ctx.JSON(http.StatusOK, configResponse)
+
+	})
+	router.POST("/buildContextStatus", func(ctx *gin.Context) {
+		pipeline, err := validateJson[zjson.Pipeline](ctx.Request.Body)
+		if err != nil {
+			log.Printf("Invalid json request; err=%v", err)
+			ctx.String(err.Status(), err.Error())
```
[the remark this answers is not in the document]
Comment (FGRCL) [reply main.go:300]: `validateJson` does enforce returning an `HTTPError` as its error. It's the same behaviour as `/pipeline`.

File: main.go [hunk 3]
```diff
@@ -283,6 +285,26 @@ func main() {
...
 	})
+	router.GET("/config", func(ctx *gin.Context) {
+		configResponse := zjson.Config{
+			IsLocal: config.IsLocal,
+		}
+
+		ctx.JSON(http.StatusOK, configResponse)
+
+	})
+	router.POST("/buildContextStatus", func(ctx *gin.Context) {
```
[the remark this answers is not in the document]
Comment (FGRCL) [reply main.go:296]: yep, replaced it with hyphens

File: executor.go [hunk 4]
```diff
@@ -707,3 +707,33 @@ func cloudExecute(pipeline *zjson.Pipeline, id int64, executionId string, build
 		return
 	}
 }
+
+func getBuildContextStatus(pipeline *zjson.Pipeline, cfg Config) *[]zjson.BuildContextStatus {
```
[the remark this answers is not in the document]
Comment (FGRCL) [reply executor.go:711]: done.

**Review by plparent — APPROVED** [review]

**Pull request merged**